Can't seem to run on 1.16.

Hey, I downloaded and dropped in the newest update for this mod and I can't get it to run.

Using:
fabric-api-0.4.32+build.296-1.16
olbm-1.0.0+1.15.2
fabric-loader-0.7.8+build.184-20w06a

<details><summary>The game crashed whilst initializing game
Error: java.lang.RuntimeException: Could not execute entrypoint stage 'client' due to errors! **Click to Expand**</summary><p>

---
---- Minecraft Crash Report ----
// I let you down. Sorry :(

Time: 2/15/20 1:33 PM
Description: Initializing game

java.lang.RuntimeException: Could not execute entrypoint stage 'client' due to errors!
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke0(EntrypointUtils.java:55)
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke(EntrypointUtils.java:34)
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointClient.start(EntrypointClient.java:33)
	at net.minecraft.class_310.<init>(class_310.java:399)
	at net.minecraft.client.main.Main.main(Main.java:166)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:193)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:138)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26)
	Suppressed: java.lang.NoClassDefFoundError: com/swordglowsblue/artifice/api/ArtificeResourcePack$ClientResourcePackBuilder
		at io.github.alloffabric.olbm.OLBClient.onInitializeClient(OLBClient.java:28)
		at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointClient$$Lambda$1499/392929911.accept(Unknown Source)
		at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke0(EntrypointUtils.java:48)
		... 11 more
	Caused by: java.lang.ClassNotFoundException: com.swordglowsblue.artifice.api.ArtificeResourcePack$ClientResourcePackBuilder
		at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
		at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
		at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
		at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
		at net.fabricmc.loader.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:161)
		at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
		... 14 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke0(EntrypointUtils.java:55)
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointUtils.invoke(EntrypointUtils.java:34)
	at net.fabricmc.loader.entrypoint.minecraft.hooks.EntrypointClient.start(EntrypointClient.java:33)
	at net.minecraft.class_310.<init>(class_310.java:399)

-- Initialization --
Details:
Stacktrace:
	at net.minecraft.client.main.Main.main(Main.java:166)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:193)
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:138)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:26)

-- System Details --
Details:
	Minecraft Version: Snapshot 20w06a
	Minecraft Version ID: 20w06a
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 1190662032 bytes (1135 MB) / 1744830464 bytes (1664 MB) up to 2147483648 bytes (2048 MB)
	CPUs: 4
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		fabric: Fabric API 0.4.32+build.296-1.16
		fabric-api-base: Fabric API Base 0.1.2+b7f9825dbf
		fabric-biomes-v1: Fabric Biomes (v1) 0.2.0+b7436ccabf
		fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.4+c6a8ea89bf
		fabric-commands-v0: Fabric Commands (v0) 0.1.2+b7f9825dbf
		fabric-containers-v0: Fabric Containers (v0) 0.1.4+9421522dbf
		fabric-content-registries-v0: Fabric Content Registries (v0) 0.1.3+b7f9825dbf
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.2+b7f9825dbf
		fabric-dimensions-v1: fabric-dimensions-v1 0.2.5+9421522dbf
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.2.8+9421522dbf
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.1.2+b7f9825dbf
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.1.6+ec40b2e1bf
		fabric-keybindings-v0: Fabric Key Bindings (v0) 0.1.1+dfdb52d6bf
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 0.1.5+e08a7305bf
		fabric-mining-levels-v0: Fabric Mining Levels (v0) 0.1.1+b7f9825dbf
		fabric-models-v0: Fabric Models (v0) 0.1.0+dfdb52d6bf
		fabric-networking-blockentity-v0: Fabric Networking Block Entity (v0) 0.2.4+9421522dbf
		fabric-networking-v0: Fabric Networking (v0) 0.1.8+9421522dbf
		fabric-object-builders-v0: Fabric Object Builders (v0) 0.1.3+e4c9a9c3bf
		fabric-particles-v1: fabric-particles-v1 0.1.1+dfdb52d6bf
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.2.7+9421522dbf
		fabric-renderer-api-v1: Fabric Renderer API (v1) 0.2.10+7d8ccbd7bf
		fabric-renderer-indigo: Fabric Renderer - Indigo 0.2.24+cfa5f8f2bf
		fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 2.0.1+5a0f9a60bf
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.1.3+b7f9825dbf
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 0.1.6+12515ed9bf
		fabric-rendering-v0: Fabric Rendering (v0) 1.1.0+53410490bf
		fabric-rendering-v1: Fabric Rendering (v1) 0.1.0+53410490bf
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.1.10+06c939b3bf
		fabric-tag-extensions-v0: Fabric Tag Extensions (v0) 0.1.3+abd91580bf
		fabric-textures-v0: Fabric Textures (v0) 1.0.4+821cdba7bf
		fabricloader: Fabric Loader 0.7.8+build.184
		jankson: Jankson 2.0.1+j1.2.0
		minecraft: Minecraft 1.16-alpha.20.6.a
		olbm: OLBM 1.0.0+1.15.2
		staticdata: Static Data 1.1.2
	Launched Version: fabric-loader-0.7.8+build.184-20w06a
	Backend library: LWJGL version 3.2.2 build 10
	Backend API: NO CONTEXT
	GL Caps: 
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	CPU: <unknown>

---

Collapsible until here.
</p></details>

--- a/build.gradle
+++ b/build.gradle
@@ -2,7 +2,7 @@ plugins {
 	id "java"
 	id "eclipse"
 	id "idea"
-	id "fabric-loom" version "0.2.6-SNAPSHOT"
+	id "fabric-loom" version "0.4.30"
 	id "maven-publish"
 	id "com.jfrog.artifactory" version "4.9.0"
 }

--- a/gradle.properties
+++ b/gradle.properties
@@ -3,9 +3,9 @@ org.gradle.jvmargs=-Xmx1G
 
 # Fabric Properties
 # check these on https://fabricmc.net/use
-minecraft_version=1.15.2
-yarn_build=3
-loader_version=0.7.6+build.179
+minecraft_version=1.16.1
+yarn_build=21
+loader_version=0.9.0+build.204
 
 # Mod Properties
 mod_version = 1.0.1
@@ -14,7 +14,7 @@ archives_base_name = olbm
 
 # Dependencies
 # currently not on the main fabric site, check on the maven: https://maven.fabricmc.net/net/fabricmc/fabric-api/fabric-api
-fabric_version=0.4.29+build.290-1.15
-artifice_version=0.6.0+1.15.2
+fabric_version=0.16.0+build.384-1.16.1
+artifice_version=0.8.1+1.16.1
 jankson_version=2.0.1+j1.2.0
 static_data_version=1.1.2

--- a/src/main/java/io/github/alloffabric/olbm/OLBM.java
+++ b/src/main/java/io/github/alloffabric/olbm/OLBM.java
@@ -6,13 +6,13 @@ import io.github.alloffabric.olbm.util.OLBRegistry;
 import net.fabricmc.api.ModInitializer;
 import net.fabricmc.fabric.api.client.itemgroup.FabricItemGroupBuilder;
 import net.fabricmc.fabric.api.container.ContainerProviderRegistry;
-import net.minecraft.inventory.BasicInventory;
+import net.minecraft.inventory.SimpleInventory;
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemGroup;
 import net.minecraft.item.ItemStack;
 import net.minecraft.item.Items;
-import net.minecraft.util.DefaultedList;
 import net.minecraft.util.Identifier;
+import net.minecraft.util.collection.DefaultedList;
 import net.minecraft.util.registry.Registry;
 import net.minecraft.util.registry.SimpleRegistry;
 import org.apache.logging.log4j.LogManager;
@@ -58,7 +58,7 @@ public class OLBM implements ModInitializer {
 				ItemStack stack = buf.readItemStack();
 				stacks.set(i, stack);
 			}
-			return new LootBagContainer(id, syncId, player, new BasicInventory(stacks.toArray(new ItemStack[]{})));
+			return new LootBagContainer(id, syncId, player, new SimpleInventory(stacks.toArray(new ItemStack[]{})));
 		});
 	}
 

--- a/src/main/java/io/github/alloffabric/olbm/inventory/LootBagContainer.java
+++ b/src/main/java/io/github/alloffabric/olbm/inventory/LootBagContainer.java
@@ -1,17 +1,17 @@
 package io.github.alloffabric.olbm.inventory;
 
-import net.minecraft.container.ContainerType;
-import net.minecraft.container.GenericContainer;
 import net.minecraft.entity.player.PlayerEntity;
 import net.minecraft.inventory.Inventory;
+import net.minecraft.screen.GenericContainerScreenHandler;
+import net.minecraft.screen.ScreenHandlerType;
 import net.minecraft.util.Identifier;
 
-public class LootBagContainer extends GenericContainer {
+public class LootBagContainer extends GenericContainerScreenHandler {
 	private PlayerEntity player;
 	private Identifier id;
 
 	public LootBagContainer(Identifier id, int syncId, PlayerEntity player, Inventory inv) {
-		super(ContainerType.GENERIC_9X3, syncId, player.inventory, inv, 3);
+		super(ScreenHandlerType.GENERIC_9X3, syncId, player.inventory, inv, 3);
 		this.id = id;
 		this.player = player;
 	}

--- a/src/main/java/io/github/alloffabric/olbm/item/LootBagItem.java
+++ b/src/main/java/io/github/alloffabric/olbm/item/LootBagItem.java
@@ -34,9 +34,9 @@ public class LootBagItem extends Item {
 		if (!world.isClient) {
 			ServerWorld server = (ServerWorld)world;
 			Identifier id = type.getTableId();
-			LootContext ctx = new LootContext.Builder(server).put(LootContextParameters.THIS_ENTITY, player).put(LootContextParameters.POSITION, player.getBlockPos()).build(LootContextTypes.GIFT);
-			LootTable supplier = server.getServer().getLootManager().getSupplier(id);
-			List<ItemStack> stacks = supplier.getDrops(ctx);
+			LootContext ctx = new LootContext.Builder(server).parameter(LootContextParameters.THIS_ENTITY, player).parameter(LootContextParameters.POSITION, player.getBlockPos()).build(LootContextTypes.GIFT);
+			LootTable supplier = server.getServer().getLootManager().getTable(id);
+			List<ItemStack> stacks = supplier.generateLoot(ctx);
 			ContainerProviderRegistry.INSTANCE.openContainer(new Identifier(OLBM.MODID, "loot_bag"), player, buf -> {
 				buf.writeIdentifier(type.getId());
 				buf.writeVarInt(stacks.size());
@@ -61,7 +61,7 @@ public class LootBagItem extends Item {
 	}
 
 	@Override
-	public boolean hasEnchantmentGlint(ItemStack stack) {
+	public boolean hasGlint(ItemStack stack) {
 		return type.hasGlint();
 	}
 

--- a/src/main/java/io/github/alloffabric/olbm/util/OLBRegistry.java
+++ b/src/main/java/io/github/alloffabric/olbm/util/OLBRegistry.java
@@ -1,13 +1,20 @@
 package io.github.alloffabric.olbm.util;
 
+import com.mojang.serialization.Lifecycle;
 import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.util.registry.RegistryKey;
 import net.minecraft.util.registry.SimpleRegistry;
 
 /**
  * Exists *purely* so we have access to a `containsId()` method on the server. Thaaaaaanks, ProGuard.
  */
 public class OLBRegistry<T> extends SimpleRegistry<T> {
+	public OLBRegistry() {
+		super(RegistryKey.ofRegistry(new Identifier("olbregistry")), Lifecycle.stable());
+	}
+
 	public boolean hasId(Identifier id) {
-		return this.entries.containsKey(id);
+		return this.entriesById.containsKey(id);
 	}
 }

